core/src/main/java/com/googlecode/psiprobe/tokenizer/Tokenizer.java

```diff
@@ -267,18 +267,17 @@ private int lookupSymbol(char chr) throws IOException {
     if (index >= 0) {
       // the index could be anywhere within a group of sybols with the same first letter
       // so we need to scroll up the group to make sure we start test from the beginning
-      while (index > 0 && ((TokenizerSymbol) symbols.get(index - 1)).compareTo(chrObj) == 0) {
+      while (index > 0 && symbols.get(index - 1).compareTo(chrObj) == 0) {
         index--;
       }
       while (index < symbols.size()) {
-        TokenizerSymbol symbol = ((TokenizerSymbol) symbols.get(index));
+        TokenizerSymbol symbol = (symbols.get(index));
```

Comment:
I'll catch those on another commit.  There are quite a few of those that Sonar picked up on.  I left alone here so I could isolate the change occurring better.